src/plugins/Wallet/real.ts

```diff
@@ -47,8 +48,37 @@ async function sendTx<R, T extends TransactionObject<R>>(txObject: T, tx: Tx = {
                     gas,
                     gasPrice,
                 })
+                /**
+                 * @returns {String}
+                 *
+                 * Is fired right after the transaction is
+                 * sent and a transaction hash is available.
+                 */
                 .on('transactionHash', (hash: string) => listeners?.onTransactionHash?.(hash))
+                /**
+                 * @returns {TxReceipt}
+                 *
+                 * Is fired when the transaction receipt is available.
+                 */
                 .on('receipt', (receipt: TxReceipt) => listeners?.onReceipt?.(receipt))
+                /**
+                 * @returns {Number, TxReceipt}
+                 *
+                 * Is fired for every confirmation up to the 12th confirmation.
+                 * Receives the confirmation number as the first and the receipt
+                 * as the second argument.
+                 *
+                 * Fired from confirmation 0 on, which is the block where its minded.
+                 */
+                .on('confirmation', (no: number, receipt: TxReceipt) => listeners?.onConfirmation?.(no, receipt))
```

Comment:
From the [White Paper][wp]:

> Ethereum implements a simplified version of GHOST which only goes down seven levels. Specifically, it is defined as follows:
> 
> + A block must specify a parent, and it must specify 0 or more uncles
> + An uncle included in block B must have the following properties:
> + It must be a direct child of the k-th generation ancestor of B, where 2 <= k <= 7.
> + ...

Dual to the possibility of uncle block, 7 confirmations are safety indeed, the `confirmation` event emits at the first `12th`(every 12th), it is closed to be confirmed.

Another ref from the [stackexchange][se].

---

The other way is adding a listener keeping requesting Ethereum till the block is totally mined, but it seems costs a lot, and the logic of wallet will be more complex.

[wp]: https://github.com/ethereum/wiki/wiki/White-Paper#modified-ghost-implementation
[se]: https://ethereum.stackexchange.com/questions/319/what-number-of-confirmations-is-considered-secure-in-ethereum